### ISSUE TITLE
2.5.3 parsing behaviour change?

### ISSUE THREAD

**Issue by stevezau** [dateutil/dateutil#258]
Hey,

Just noticed my tests were failing on http://github.com/Flexget/Flexget

We use dateutil heavily, it looks like the latest change is causing "01 02 03" (assuming year first) to be parsed as 2001-03-02

Previously it was parsed as 2001-03-02?

**Comment by pganssle** [comment]
I don't think any actual behavior should have changed from 2.5.2 to 2.5.3. The only change is in regards to unambiguous dates like `'2001-03-02'`, which were failing when `dayfirst` was specified.

Are you passing any arguments to the parser? There's more detail about the behavior in Issue #238 (which should probably go into the documentation, now that I think about it).

FWIW:

```python
>>> from dateutil import __version__
>>> from dateutil import parser
>>> __version__
'2.5.3'
>>> parser.parse('010203')
datetime.datetime(2003, 1, 2, 0, 0)
```

**Comment by pganssle** [comment]
@stevezau By the way, I noticed that in your latest commit on FlexGet you seem to think that you need to drop dateutil 2.5.3 due to #233, but that bug is fixed in 2.5.3. The current behavior is going to be the behavior of the parser from now on.

There will likely be an interface in version 2.6.0 that allows you to specify more "fallback" options, but you should probably program against the current interface rather than relying on the old behavior.

**Comment by stevezau** [comment]
@pganssle yeah I commited a change to ignore the last 2 releases as an attempt to not effect our users.

When i get a spare second i'll have a proper look into what's going on.
